lib/truffle/truffle/patching.rb

```diff
@@ -1,140 +1,68 @@
-require 'pathname'
-
 module Truffle::Patching
-  DIR     = Pathname(Truffle::Boot.ruby_home).join("lib/patches")
-  # Allowed operations are: :before, :after, :instead, :ignore
-  PATCHES = { "bundler"                                 => :before,
-              "bundler/version"                         => :after,
-              "bundler/cli/exec"                        => :after,
-              "bundler/compact_index_client/updater"    => :after,
-              "bundler/current_ruby"                    => :after,
-              "bundler/dependency"                      => :after,
-              "bundler/fetcher/downloader"              => :after,
-              "bundler/fetcher/compact_index"           => :after,
-              "bundler/ruby_version"                    => :after,
-              "bundler/source/rubygems"                 => :after,
-
-              "rubygems"                                => :before,
-              "rubygems/ext"                            => :after,
-              "rubygems/package"                        => :after,
-              "rubygems/remote_fetcher"                 => :after,
-              "rubygems/request"                        => :after,
-              "rubygems/request_set/gem_dependency_api" => :after,
-
-              "rspec/support"                           => :after,
-
-              "unf/normalizer_cruby"                    => :before,
-  }
-
-  PATCHES.each do |file, _|
-    unless DIR.join(file).sub_ext('.rb').exist?
-      raise "file #{file} is missing in #{DIR}"
+  def patches
+    @patches ||= begin
+      require 'pathname'
```

Comment:
Ah right it does call it for did_you_mean, it does single hash access and returns. I was thinking about activations with patching.